mathesar_ui/src/stores/table-data/display.ts

```diff
@@ -0,0 +1,88 @@
+import { writable, get } from 'svelte/store';
+import type { Writable, Unsubscriber } from 'svelte/store';
+import type { TabularType, DBObjectEntry } from '@mathesar/App.d';
+import type { Meta } from './meta';
+import type { Columns, TableColumn } from './columns';
+
+export interface ColumnPosition {
+  width: number,
+  left: number
+}
+export type ColumnPositionMap = Map<string, ColumnPosition>;
+
+export const ROW_CONTROL_COLUMN_WIDTH = 70;
+export const GROUP_MARGIN_LEFT = 30;
+export const DEFAULT_ROW_RIGHT_PADDING = 100;
+export const DEFAULT_COLUMN_WIDTH = 160;
+
+function recalculateColumnPositions(columnPositionMap: ColumnPositionMap, columns: TableColumn[]) {
+  let left = ROW_CONTROL_COLUMN_WIDTH;
+  const newColumnPositionMap: ColumnPositionMap = new Map(columnPositionMap);
```

Comment:
Hey @pavish, fabulous work here! Quick question: What's the reasoning for passing ` columnPositionMap`  inside of `new  Map()` ? It prevents `columnPositionMap` from updating correctly in line 76-78: ```
this.columnPositionMap.update(
        (map) => recalculateColumnPositions(map, columnData.data),
      ); 
``` when calling `columns.fetch() ` after a column is deleted.

Reply:
@mr-gabe49 When a column has been previously resized, the columnPositionMap would contain a different value for width instead of the default column width. We create a new map using the existing map to retain it.

Good catch, I hadn't considered the effect of deletion.

In your PR for column deletion, we could update lines 30-32 to:
```
  const newColumnPositionMap: ColumnPositionMap = new Map();
  columns.forEach((column) => {
    const columnWidth = columnPositionMap.get(column.name)?.width;
```
Since we have access to the old map, we could use it to get the previous width. This would handle position recalc for both resize and delete.